sql: SHOW CREATE ALL TRIGGERS

OriSavir opened the pull request:
Adding SHOW CREATE ALL TRIGGERS.
Includes parser syntax additions and generator logic changes.
Additionally included adding the `crdb_interal.create_trigger_statements` table
and implementing table_id as a virtual index (since trigger_id are unique only within tables).


Release note (sql): SHOW CREATE ALL TRIGGERS is now a valid query with correct output.
Epic: CRDB-49582

Resolves #144709

OriSavir committed:
sql: add `SHOW CREATE ALL TRIGGERS`
and implement it with optimized query.

Implemented `SHOW CREATE ALL TRIGGERS` command.
This consists on parser changes (sql.y), generator functions,
addition of the command to delegator, and also creation of
`crdb_internal.create_trigger_statements` as a virtual table.
The generators query the table, which is indexed on `table_id`
and the query result contains the create statements of all triggers.
This is a useful change for the task of a `SHOW CREATE` statement
that returns create statements for all objects of the database.

Release note (sql): Add `SHOW CREATE ALL TRIGGERS` as a command
to CockroachDB

Epic: CRDB-49582

yuzefovich reviewed (commented):
Nice work! I took a cursory look and left a few nits, you'll also need to run `./dev gen docs` to regenerate some things not related to your change.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-:-OSKZLJkCdCpKaSi8q7r:b-s31m8g)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[`pkg/sql/opt/exec/explain/plan_gist_factory.go` line 188 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSKZbH1851kzC6PGIXr:-OSKZbH1851kzC6PGIXs:bczw8ja) ([raw file](https://github.com/cockroachdb/cockroach/blob/b307a8da0f8afe3adb3c77c09afae7fe4197c92d/pkg/sql/opt/exec/explain/plan_gist_factory.go#L188)):*
> ```Go
> 			// possible because the code does not update shared state and does
> 			// not manipulate locks.
> 			fmt.Printf("recovered from panic in DecodePlanGistToRows: %v\n", r)
> ```

nit: this debugging info should be removed.
___
*[`pkg/sql/conn_executor_exec.go` line 3428 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSKZWtl3K9wtBznLhcT:-OSKZWtl3K9wtBznLhcU:bczw8ja) ([raw file](https://github.com/cockroachdb/cockroach/blob/b307a8da0f8afe3adb3c77c09afae7fe4197c92d/pkg/sql/conn_executor_exec.go#L3428)):*
> ```Go
> 		}
> 		planStr := ih.planGist.String()
> 		fmt.Printf("plan gist before decode: %s\n", planStr)
> ```

nit: this debugging info should be removed.
___
*[`pkg/util/buildutil/crdb_test_off.go` line 16 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSKZdfv7O269bKy_JTE:-OSKZdfv7O269bKy_JTF:be4019w) ([raw file](https://github.com/cockroachdb/cockroach/blob/b307a8da0f8afe3adb3c77c09afae7fe4197c92d/pkg/util/buildutil/crdb_test_off.go#L16)):*
> ```Go
> // metamorphic-style perturbations that will not affect test results but will
> // exercise different parts of the code.
> const CrdbTestBuild = true
> ```

nit: this debugging change should be removed.
___
*[`pkg/sql/crdb_internal.go` line 4137 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSKYyWVDfy0jLi8ph8m:-OSKYyWVDfy0jLi8ph8n:be0g6ts) ([raw file](https://github.com/cockroachdb/cockroach/blob/b89d3e257cfe6bfef5e936749bab0e91e7bfe939/pkg/sql/crdb_internal.go#L4137)):*
> ```Go
> 		},
> 		// Index on (table_id, trigger_id)
> 		//TODO: Can you indes by two columns with virtual tables?
> ```

No, this is currently not possible - we'd hit an assertion here https://github.com/cockroachdb/cockroach/blob/d2f53d7487175893289d2ed4df3b70270652fe7b/pkg/sql/virtual_schema.go#L822-L825.


<!-- Sent from Reviewable.io -->

OriSavir commented:
Thanks for the comments @yuzefovich, pushed in fixes to that and some more!

rafiss reviewed (dismissed):
nice job getting this working! let me know if you have any questions about the comments i left.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-:-OSQNZDe81lCqHu5IGkp:bxrn79a)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @OriSavir and @yuzefovich)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQHLFy547buDYlnFRu:-OSQHLFy547buDYlnFRv:bfb6l2f):*
the commit should have a package name here in the prefix, and have a more clear title. so this could be: `sql: add syntax for SHOW CREATE ALL TRIGGERS and implement it`
___
*[`-- commits` line 8 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQGwRt4SPdHTp7nyzs:-OSQGwRt4SPdHTp7nyzt:b-c9e3ws):*
the release note should be written with complete sentences that a user could understand. also this note says "tables" instead of "triggers."

in this case, it could be "Added syntax for the `SHOW CREATE ALL TRIGGERS` command, which can be used to show CREATE statements for all triggers in the current database.
___
*[`-- commits` line 11 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQGtd-Dz0f02NePsVe:-OSQGtd-Dz0f02NePsVf:b4gjsvk):*
remove these extra lines from the commit message.
___
*[`-- commits` line 28 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQJB0P5j0Cawfch8Lu:-OSQJB0P5j0Cawfch8Lv:bs3oxn9):*
Let's squash the two commits into one. The first commit does not stand on its own, and doesn't capture a clear "unit of work," so for the sake of maintaining a readable commit history, it would be better to combine both of them.
___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 30 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQL-DC2ZXXiCy9kFRJ:-OSQL-DC2ZXXiCy9kFRK:b-dj9bu) ([raw file](https://github.com/cockroachdb/cockroach/blob/6537013178627bf73b999c88e334e53a6458654f/pkg/sql/sem/builtins/show_create_all_triggers.go#L30)):*
> ```Go
> 	ctx context.Context, evalPlanner eval.Planner, txn *kv.Txn, dbName string, acc *mon.BoundAccount,
> ) (triggerIds []triggerIdPair, retErr error) { //TODO: Check up on ID field names etc
> 	//TODO: Want to get (trigger_id, table_id)
> ```

nit: remove this TODO
___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 72 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQLNB79RweoZijwFYM:-OSQLNB79RweoZijwFYN:b-jl29yz) ([raw file](https://github.com/cockroachdb/cockroach/blob/6537013178627bf73b999c88e334e53a6458654f/pkg/sql/sem/builtins/show_create_all_triggers.go#L72)):*
> ```Go
> 	dbName string,
> ) (_ tree.Datum, err error) {
> 	query := fmt.Sprintf(`
> ```

it would be very useful to add a comment here saying that we query by tableID since `crdb_internal.create_trigger_statements` only has an index on tableID. and also mention that this means we need to iterate through the results to find the row with the trigger ID we are looking for.
___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 100 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQLgSg-713LR1JQ3Ny:-OSQLgSg-713LR1JQ3Nz:bpf69mh) ([raw file](https://github.com/cockroachdb/cockroach/blob/6537013178627bf73b999c88e334e53a6458654f/pkg/sql/sem/builtins/show_create_all_triggers.go#L100)):*
> ```Go
> 		row := iter.Cur()
> 		if len(row) != 2 {
> 			return nil, errors.Newf("expected 2 columns in result, got %d", len(row))
> ```

nit: use `errors.AssertionFailedf()` for this one
___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 108 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQLqur8LYm9CKPD-TS:-OSQLqur8LYm9CKPD-TT:bkjp5z3) ([raw file](https://github.com/cockroachdb/cockroach/blob/6537013178627bf73b999c88e334e53a6458654f/pkg/sql/sem/builtins/show_create_all_triggers.go#L108)):*
> ```Go
> 	}
> 
> 	if err != nil {
> ```

is this check needed? which `err` is it checking?
___
*[`pkg/sql/sem/builtins/generator_builtins.go` line 586 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQId03793xePiNiX9_:-OSQId03793xePiNiX9a:b4dsp62) ([raw file](https://github.com/cockroachdb/cockroach/blob/b89d3e257cfe6bfef5e936749bab0e91e7bfe939/pkg/sql/sem/builtins/generator_builtins.go#L586)):*
> ```Go
> 			showCreateAllTriggersGeneratorType,
> 			makeShowCreateAllTriggersGenerator,
> 			`Returns rows of CREATE trigger statements.
> ```

nit: don't add a newline here.
___
*[`pkg/sql/parser/testdata/show_create_all_triggers` line 2 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQMYe46PKRcfvt4QeQ:-OSQMYe46PKRcfvt4QeR:b8bjktk) ([raw file](https://github.com/cockroachdb/cockroach/blob/6537013178627bf73b999c88e334e53a6458654f/pkg/sql/parser/testdata/show_create_all_triggers#L2)):*
> ```
> parse
> SHOW CREATE ALL TRIGGERS
> ```

there's no need to create a new test file for this. the test you added in the `show` file is sufficient
___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 17 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQIn-48KoUxtP3GvQ1:-OSQIn-48KoUxtP3GvQ2:b-uqn9y4) ([raw file](https://github.com/cockroachdb/cockroach/blob/b89d3e257cfe6bfef5e936749bab0e91e7bfe939/pkg/sql/sem/builtins/show_create_all_triggers.go#L17)):*
> ```Go
> )
> 
> type triggerIdPair struct {
> ```

let's find a better name for this. instead of `triggerIdPair`, how about we call this `tableTriggerPair`?
___
*[`pkg/sql/crdb_internal.go` line 3947 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQN-6B2uZCkELdzCZP:-OSQN-6B2uZCkELdzCZQ:bd6rf9a) ([raw file](https://github.com/cockroachdb/cockroach/blob/6537013178627bf73b999c88e334e53a6458654f/pkg/sql/crdb_internal.go#L3947)):*
> ```Go
> }
> 
> func renderCreateTriggerStatement(
> ```

the reason for introducing this function was so that we could use it in two places: the crdb_internal table you just added, and the existing code here: https://github.com/cockroachdb/cockroach/blob/5252f53191642cd6d86cbd43dc88494d036e44dd/pkg/sql/show_trigger.go#L90

please update that usage to use this function. also, it would be better to define `renderCreateTriggerStatement` in that file instead of here, to keep the code more organized. 

this change can be addressed as a separate commit (in this same PR), since it's just a refactor to reduce duplication.


<!-- Sent from Reviewable.io -->

OriSavir committed:
sql: refactor render function for `ShowCreateTrigger()`

Moved `renderCreateTriggerStatement()` into
`show_trigger.go` and used it in `ShowCreateTrigger()`
to avoid repeating the code for generating trigger
create statements.
Also, removes the work-around explicitly
iterating over query, as the SQL query execution
bug is fixed.

Release note: None
Epic: CRDB-49582

fqazi reviewed (changes requested):
> Reviewed 7 of 13 files at r1, 33 of 40 files at r4, 1 of 1 files at r5, 4 of 4 files at r6.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/147836)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @OriSavir and @rafiss)

___
*[`pkg/sql/crdb_internal.go` line 3947 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQN-6B2uZCkELdzCZP:-OSUT3h24MCYEA_A6woQ:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6537013178627bf73b999c88e334e53a6458654f/pkg/sql/crdb_internal.go#L3947)):*
<details><summary><i>Previously, OriSavir (Oriel Savir) wrote…</i></summary><blockquote>

Pushed some changes addressing these, thanks!
</blockquote></details>

Done.
___
*[`pkg/sql/sem/builtins/generator_builtins.go` line 586 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQId03793xePiNiX9_:-OSUT9W38Fv7jJCva2uV:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b89d3e257cfe6bfef5e936749bab0e91e7bfe939/pkg/sql/sem/builtins/generator_builtins.go#L586)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

nit: don't add a newline here.
</blockquote></details>

Done.
___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 17 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQIn-48KoUxtP3GvQ1:-OSUTBRs2bVbPNMVmUAR:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b89d3e257cfe6bfef5e936749bab0e91e7bfe939/pkg/sql/sem/builtins/show_create_all_triggers.go#L17)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

let's find a better name for this. instead of `triggerIdPair`, how about we call this `tableTriggerPair`?
</blockquote></details>

Done.
___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 30 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQL-DC2ZXXiCy9kFRJ:-OSUTCJ82k6tp_R0kFBG:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6537013178627bf73b999c88e334e53a6458654f/pkg/sql/sem/builtins/show_create_all_triggers.go#L30)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

nit: remove this TODO
</blockquote></details>

Done.
___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 100 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQLgSg-713LR1JQ3Ny:-OSUTfoL-7OzMtDRGLR_:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6537013178627bf73b999c88e334e53a6458654f/pkg/sql/sem/builtins/show_create_all_triggers.go#L100)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

nit: use `errors.AssertionFailedf()` for this one
</blockquote></details>

Done.
___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 108 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQLqur8LYm9CKPD-TS:-OSUTnAKBXMiAw2f7m8K:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6537013178627bf73b999c88e334e53a6458654f/pkg/sql/sem/builtins/show_create_all_triggers.go#L108)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

is this check needed? which `err` is it checking?
</blockquote></details>

Done.
___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 74 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSUTbyh6YXoHzePYghv:-OSUTbyh6YXoHzePYghw:bpxhchs) ([raw file](https://github.com/cockroachdb/cockroach/blob/0216356a5690e3fc84c22f467d1399614115c6b9/pkg/sql/sem/builtins/show_create_all_triggers.go#L74)):*
> ```Go
> 	// only has `table_id` as an index, due to it not currently being possible to index by
> 	// two columns on a virtual table. It's worth noting that
> 	// trigger IDs are only unique within a table, but not between tables.
> ```

Is this still disallowed after the queries bug fix?


<!-- Sent from Reviewable.io -->

OriSavir reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/147836)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @fqazi, @rafiss, and @yuzefovich)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQHLFy547buDYlnFRu:-OSUX3Hy7qrcpUB9PbuL:b-896fix):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

the commit should have a package name here in the prefix, and have a more clear title. so this could be: `sql: add syntax for SHOW CREATE ALL TRIGGERS and implement it`
</blockquote></details>

Done.
___
*[`-- commits` line 8 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQGwRt4SPdHTp7nyzs:-OSUX20BCr9LsabQ-TJL:b-896fix):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

the release note should be written with complete sentences that a user could understand. also this note says "tables" instead of "triggers."

in this case, it could be "Added syntax for the `SHOW CREATE ALL TRIGGERS` command, which can be used to show CREATE statements for all triggers in the current database.
</blockquote></details>

Done.
___
*[`-- commits` line 11 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQGtd-Dz0f02NePsVe:-OSUX1eT-cLPQjdUvbpY:b-896fix):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

remove these extra lines from the commit message.
</blockquote></details>

Done.
___
*[`-- commits` line 28 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQJB0P5j0Cawfch8Lu:-OSUX3np8SAGoHt_HXe3:b-896fix):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

Let's squash the two commits into one. The first commit does not stand on its own, and doesn't capture a clear "unit of work," so for the sake of maintaining a readable commit history, it would be better to combine both of them.
</blockquote></details>

Done.
___
*[`pkg/sql/conn_executor_exec.go` line 3428 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSKZWtl3K9wtBznLhcT:-OSUX5DlEYkj3QxDkApr:bylqgbp) ([raw file](https://github.com/cockroachdb/cockroach/blob/b307a8da0f8afe3adb3c77c09afae7fe4197c92d/pkg/sql/conn_executor_exec.go#L3428)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: this debugging info should be removed.
</blockquote></details>

Done!
Thanks for catching
___
*[`pkg/sql/crdb_internal.go` line 4137 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSKYyWVDfy0jLi8ph8m:-OSUWzPP2gsksYDW40PS:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b89d3e257cfe6bfef5e936749bab0e91e7bfe939/pkg/sql/crdb_internal.go#L4137)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

No, this is currently not possible - we'd hit an assertion here https://github.com/cockroachdb/cockroach/blob/d2f53d7487175893289d2ed4df3b70270652fe7b/pkg/sql/virtual_schema.go#L822-L825.
</blockquote></details>

Done.
___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 17 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQIn-48KoUxtP3GvQ1:-OSUXG_34wNC0K2w0XJ9:b-896fja) ([raw file](https://github.com/cockroachdb/cockroach/blob/b89d3e257cfe6bfef5e936749bab0e91e7bfe939/pkg/sql/sem/builtins/show_create_all_triggers.go#L17)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done!
___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 72 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQLNB79RweoZijwFYM:-OSUXpTW7VJ3kFmCAy0z:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6537013178627bf73b999c88e334e53a6458654f/pkg/sql/sem/builtins/show_create_all_triggers.go#L72)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

it would be very useful to add a comment here saying that we query by tableID since `crdb_internal.create_trigger_statements` only has an index on tableID. and also mention that this means we need to iterate through the results to find the row with the trigger ID we are looking for.
</blockquote></details>

Done.
___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 108 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQLqur8LYm9CKPD-TS:-OSUXr0LESde4kIKpXRl:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6537013178627bf73b999c88e334e53a6458654f/pkg/sql/sem/builtins/show_create_all_triggers.go#L108)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.
___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 74 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSUTbyh6YXoHzePYghv:-OSUXNBN63XMPuCoVPmX:b-oqi2ub) ([raw file](https://github.com/cockroachdb/cockroach/blob/0216356a5690e3fc84c22f467d1399614115c6b9/pkg/sql/sem/builtins/show_create_all_triggers.go#L74)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Is this still disallowed after the queries bug fix?
</blockquote></details>

Done. The review from @yuzefovich was given after the queries bug fix and it seems to confirm that's still the case. If that changed (or changes), happy to address in a second PR today
___
*[`pkg/sql/opt/exec/explain/plan_gist_factory.go` line 188 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSKZbH1851kzC6PGIXr:-OSUXt8P7H8UqI9pCuvl:b-896fja) ([raw file](https://github.com/cockroachdb/cockroach/blob/b307a8da0f8afe3adb3c77c09afae7fe4197c92d/pkg/sql/opt/exec/explain/plan_gist_factory.go#L188)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: this debugging info should be removed.
</blockquote></details>

Done!
___
*[`pkg/sql/parser/testdata/show_create_all_triggers` line 2 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSQMYe46PKRcfvt4QeQ:-OSUXy09E3KvTQLAlLjJ:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6537013178627bf73b999c88e334e53a6458654f/pkg/sql/parser/testdata/show_create_all_triggers#L2)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

there's no need to create a new test file for this. the test you added in the `show` file is sufficient
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-:-OSUZAnTB1S_njxpDd1i:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @rafiss and @yuzefovich)

___
*[`pkg/sql/sem/builtins/show_create_all_triggers.go` line 74 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/147836#-OSUTbyh6YXoHzePYghv:-OSUYygq8xqxMCbmfYbi:b-q3ep1k) ([raw file](https://github.com/cockroachdb/cockroach/blob/0216356a5690e3fc84c22f467d1399614115c6b9/pkg/sql/sem/builtins/show_create_all_triggers.go#L74)):*
<details><summary><i>Previously, OriSavir (Oriel Savir) wrote…</i></summary><blockquote>

Done. The review from @yuzefovich was given after the queries bug fix and it seems to confirm that's still the case. If that changed (or changes), happy to address in a second PR today
</blockquote></details>

Ah fair


<!-- Sent from Reviewable.io -->

OriSavir commented:
bors r+

OriSavir commented:
bors r+
